lib/jobs/install-os.js

```diff
@@ -245,10 +249,8 @@ function installOsJobFactory(
         }).then(function() {
             self._subscribeRequestProfile(function() {
                 var description = "Reboot suceeded, starting kernel download.";
-                return self.updateProgress(description, 1)
-                .then(function(){
-                    return self.profile;
-                });
+                self.updateProgress(description, 1)
+                return self.profile;
```

Comment:
There is a special feature for publish task/graph progress event is that, it will swallow the error if the publishing itself has some error, so that the event publishing will not impact the existing logic.